File: 백준/Gold/12865. 평범한 배낭/평범한 배낭.py
import sys
input = sys.stdin.readline

n, k = map(int, input().split())
stuff = []
for _ in range(n):
    w, v = map(int, input().split())
    stuff.append((w, v))
stuff.sort(key=lambda x:x[0])

dp = [0]*(k+1)
for cur_weight, cur_value in stuff:
    for weight in range(k, 0, -1):
        if cur_weight <= weight:
            dp[weight] = max(dp[weight], dp[weight - cur_weight] + cur_value)

print(max(dp))

File: 백준/Gold/12865. 평범한 배낭/평범한 배낭.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt').toString().split('\n')

const [N, K] = input[0].split(' ').map(Number)
const items = []
for (let i = 1; i < N + 1; i++) {
  items.push(input[i].split(' ').map(Number))
}

const bags = Array(K + 1).fill(0)
for (const [W, V] of items) {
  for (let i = K; i > 0; i--) {
    if (0 <= i - W) {
      bags[i] = Math.max(bags[i], bags[i - W] + V)
    }
  }
}
console.log(bags[K])